x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview_empty.test.tsx

```diff
@@ -28,6 +28,10 @@ describe('EmptyEngineOverview', () => {
     expect(wrapper.find(EuiButton).prop('href')).toEqual(`${docLinks.appSearchBase}/index.html`);
   });
 
+  it('contains FlashMessages', () => {
+    expect(wrapper.find(FlashMessages)).toHaveLength(1);
+  });
+
```

Comment:
I personally don't think we need to add this test. Unit testing React views doesn't really follow the same logic/flow you would for a pure function, but IMO checking that a certain child component exists should be done under the following circumstances:

1. Under the first `it('renders')` test, and usually only checking for "landmark" children that define the component.
For example, would I manually check every single small EuiButton or irrelevant EuiText? Absolutely not. But if my component is called something like `CurationsModal`, would I logically check for `EuiModal` under the initial it renders test? Yep.

2. If the child component is conditionally rendering and you're asserting that it did not & then did render

3. If the child component was passed as a prop and you want to quickly confirm the prop correctly rendered (e.g., `<SomeComponent header={<SomeHeader/>}><SomeChild /></SomeComponent>`)

In a scenario like this where `<FlashMessages />` is dropped in with no accompanying logic and it handles its own visibility with unit tests already written for that state, then I don't think all tests that have `<FlashMessages />` need to assert for it in a shallow test (especially since no deep diving/rendering is actually going on).

If we're not making deliberate choices about what children we check for in our `it('renders')` test, we might as well be using [snapshot testing](https://jestjs.io/docs/en/snapshot-testing) to automatically check all content instead of trying to do so manual. And to be honest, I'm a little skeptical of snapshot tests - I don't _hate_ them, but they add a decent amount of overhead/noise in diffs and I haven't found significant value in them over 1 or 2 deliberate assertions.